printer/ast_printer/ast_printer.go

```diff
@@ -0,0 +1,177 @@
+package ast_printer
+
+import (
+	"strings"
+
+	ast_pb "github.com/unpackdev/protos/dist/go/ast"
+	"github.com/unpackdev/solgo/ast"
+	"go.uber.org/zap"
+)
+
+const INDENT_SIZE = 2
+
+// Print is a function that prints the AST nodes to source code
+func Print(node ast.Node[ast.NodeType]) (string, bool) {
+	sb := strings.Builder{}
+	success := PrintRecursive(node, &sb, 0)
+	return sb.String(), success
+}
+
+// PrintRecursive is a function that prints the AST nodes to source code recursively
+func PrintRecursive(node ast.Node[ast.NodeType], sb *strings.Builder, depth int) bool {
```

Comment:
The difference between `PrintRecursive` and `Print`, is that `PrintRecursive` takes in a `*strings.Builder` and writes to it, while `Print` just takes in a node and returns a string

Reply:
Maybe i should have named it better...

Reply:
Sounds good to me, the naming itself as it's already ast_printer so you basically print recursive builder.

Reply:
We can always revisit this later on.